src/main/java/org/skriptlang/skript/lang/condition/CompoundConditional.java

```diff
@@ -0,0 +1,138 @@
+package org.skriptlang.skript.lang.condition;
+
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link Conditional} that is built of other {@link Conditional}s.
+ * It is composed of an ordered {@link Set} of {@link Conditional}s that are acted upon by a single {@link Operator}.
+ * @param <T> The context class to use for evaluation.
+ * @see ConditionalBuilder
+ */
+public class CompoundConditional<T> implements Conditional<T> {
+
+	private final LinkedHashSet<Conditional<T>> componentConditionals = new LinkedHashSet<>();
+	private final Operator operator;
+
+	/**
+	 * Whether a cache will be used during evaluation. True if this compound contains other compounds, otherwise false.
+	 */
+	private boolean useCache;
+
+	/**
+	 * @param operator The {@link Operator} to use to combine the conditionals. If {@link Operator#NOT} is used,
+	 * 					{@code conditionals} must exactly 1 conditional.
+	 * @param conditionals A collection of conditionals to combine using the operator. Must be >= 1 in length,
+	 *                     or exactly 1 if {@link Operator#NOT} is used.
+	 */
+	public CompoundConditional(Operator operator, @NotNull Collection<Conditional<T>> conditionals) {
+		if (conditionals.isEmpty())
+			throw new IllegalArgumentException("CompoundConditionals must contain at least 1 component conditional.");
+		if (operator == Operator.NOT && conditionals.size() != 1)
```

Comment:
```suggestion
		if (operator == Operator.NOT && conditionals.size() > 1)
```
does the same but makes it more obvious that having >1 conditions with a NOT is bad (same as done on L#88)

Reply:
I went with != because I think it's more clear that only 1 is allowed, no other value. With > 1 you have to reference the above if to see that 0 is also forbidden, and then also remember that size can't be negative.

Reply:
fair